woocommerce/woocommerce
# Fix/ Count() warning/error on invalid data supplied by the filter `woocommerce_cart_shipping_packages`

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This suppresses a warning on PHP <=7 / error on PHP 8+ on invalid data being sent to `woocommerce_cart_shipping_packages` filter on the StoreAPI CartController that would try to use count() without checking data type against an object implementing Countable.

Closes #42410 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add the following filter:
```php
add_filter('woocommerce_cart_shipping_packages', function() {
    return "Sorry, you cannot count on me.";
} );
```
3. Add an item to the cart and visit the Cart Page
4. On PHP >=8 no critical error should occur. 



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

This suppresses a warning/error on invalid data being sent to `woocommerce_cart_shipping_packages` filter on the StoreAPI CartController that would try to use count() without checking data type against an object implementing Countable. No need to communicate changelog.

</details>
